### PR TITLE
auth: Reduce the number of temporary memory allocations

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

This PR <strike>is a work-in-progress to reduce</strike> reduces the number of temporary memory allocations and copies in the authoritative server. Comments welcome!

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
#8370 is somewhat related. I just rebased it.

**Comment by rgacogne** [comment]
Rebased to fix conflicts.

**Comment by Habbie** [comment]
> This PR is a work-in-progress to reduce the number of temporary memory allocations and copies in the authoritative server. Comments welcome!

It was marked 'ready for review', I trust it is no longer work-in-progress?

**Comment by rgacogne** [comment]
> It was marked 'ready for review', I trust it is no longer work-in-progress?

Yes, I should have added a comment! I have a few remaining ideas but I can always open a separate PR later :)

**Comment by Habbie** [comment]
> Yes, I should have added a comment! I have a few remaining ideas but I can always open a separate PR later :)

I just edit the original comment in such cases. Thanks!

File: pdns/packethandler.cc [hunk 1]
```diff
@@ -944,18 +946,21 @@ int PacketHandler::processNotify(const DNSPacket& p)
 static bool validDNSName(const DNSName &name)
 {
```
Comment (Habbie) [inline pdns/packethandler.cc]: with how much this function now assumes about how `DNSName` works, perhaps we should move it into the class at some point? Doesn't have to be in this PR, just a thought :)

Comment (rgacogne) [reply pdns/packethandler.cc]: Yes, that would make a lot of sense!

**Comment by rgacogne** [comment]
Rebased. I'll move the `validDNSName` fonction into the `DNSName` class, but I don't mind if that happens outside of this merge request.

**Comment by rgacogne** [comment]
>  I'll move the validDNSName fonction into the DNSName class, but I don't mind if that happens outside of this merge request.

Done!

**Comment by rgacogne** [comment]
Rebased to fix the `PG sql` tests issue (see #8833).

**Pull request merged**